contracts/Consensus.sol

```diff
@@ -0,0 +1,142 @@
+pragma solidity ^0.4.24;
+
+import "./abstracts/ValidatorSet.sol";
+import "./ConsensusStorage.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract Consensus is ConsensusStorage, ValidatorSet {
+  using SafeMath for uint256;
+
+  event ChangeFinalized(address[] newSet);
+
+  modifier notFinalized() {
+    require (!isFinalized());
+    _;
+  }
+
+  modifier onlySystem() {
+    require(msg.sender == systemAddress());
+    _;
+  }
+
+  function initialize(uint256 _minStake, address _initialValidator) public returns(bool){
+    require(!isInitialized());
+    setSystemAddress();
```

Comment:
I would put the hard coded system address here as a parameter instead of inside the function.
Or maybe as a const at the beginning of the contract.